Mathlib/Geometry/Euclidean/Incenter.lean

```diff
@@ -0,0 +1,622 @@
+/-
+Copyright (c) 2025 Joseph Myers. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joseph Myers
+-/
+import Mathlib.Geometry.Euclidean.SignedDist
+import Mathlib.Geometry.Euclidean.Sphere.Basic
+
+/-!
+# Incenters and excenters of simplices.
+
+This file defines the insphere and exspheres of a simplex (tangent to the faces of the simplex),
+and the center and radius of such spheres.
+
+## Main definitions
+
+* `Affine.Simplex.ExcenterExists` says whether an excenter exists with a given set of indices
+  (that determine, up to negating all the signs, which vertices of the simplex lie on the same
+  side of the opposite face as the excenter and which lie on the opposite side of that face).
+* `Affine.Simplex.excenterWeights` are the weights of the excenter with the given set of
+  indices, if it exists, as an affine combination of the vertices.
+* `Affine.Simplex.exsphere` is the exsphere with the given set of indices, if it exists.
+* `Affine.Simplex.insphere` is the insphere with the given set of indices.
+* `Affine.Simplex.excenter` is the excenter with the given set of indices, if it exists.
+* `Affine.Simplex.incenter` is the incenter with the given set of indices.
+* `Affine.Simplex.exradius` is the exradius with the given set of indices, if it exists.
+* `Affine.Simplex.inradius` is the inradius with the given set of indices.
+
+## References
+
+* https://en.wikipedia.org/wiki/Incircle_and_excircles
+* https://en.wikipedia.org/wiki/Incenter
+
+-/
+
+
+open EuclideanGeometry
+open scoped Finset RealInnerProductSpace
+
+variable {V P : Type*} [NormedAddCommGroup V] [InnerProductSpace ℝ V] [MetricSpace P]
+variable [NormedAddTorsor V P]
+
+noncomputable section
+
+namespace Affine
+
+namespace Simplex
+
+variable {n : ℕ} [NeZero n] (s : Simplex ℝ P n)
+
+open scoped Classical in
+/-- The unnormalized weights of the vertices in an affine combination that gives an excenter with
+signs determined by the given set of indices (for the empty set, this is the incenter; for a
+singleton set, this is the excenter opposite a vertex).  An excenter with those signs exists if
+and only if the sum of these weights is nonzero (so the normalized weights sum to 1). -/
+def excenterWeightsUnnorm (signs : Set (Fin (n + 1))) (i : Fin (n + 1)) : ℝ :=
```

Comment:
My inclination here would be to use a `Finset` to avoid needing the `open scoped Classical`.

Reply:
Changed to use `Finset`.